src/anim/binder/default-anim-binder.js

```diff
@@ -79,6 +79,11 @@ class DefaultAnimBinder {
             },
 
             'weight': function (node, weightName) {
+                if (weightName.indexOf('name.') === 0) {
+                    weightName = weightName.replace('name.', '');
+                } else {
+                    weightName = Number(weightName);
```

Comment:
worth checking for NaN here?

Reply:
This will always be the index of the morph target as a string.